packages/contracts-bedrock/test/invariants/echidna/superc20/PROPERTIES.md

```diff
@@ -0,0 +1,41 @@
+supERC20 properties
+===================
+
+Valid state
+-----------
+
+| id    | description                                                           | halmos | echidna |
+| ----- | -----                                                                 | -----  | -----   |
+|   0   | calls to sendERC20 succeed as long as caller has enough balance       |        |         |
+
+Variable transition
+-------------------
+
+| id    | description                                                           | halmos | echidna |
+| ----- | -----                                                                 | -----  | -----   |
+|   1   | sendERC20 decreases the token's totalSupply in the source chain       |        |         |
+|   2   | relayERC20 increases the token's totalSupply in the destination chain |        |         |
+|   3   | only calls to `convert()` can increase the total supply across chains |        |         |
+
+High level
+----------
+
+| id    | description                                                                                                  | halmos | echidna |
+| ----- | -----                                                                                                        | -----  | -----   |
+|   4   | sum of total supply across all chains is always `<=` to `convert()`ed amount                                 |        |         |
+|   5   | tokens `sendERC20`-ed to a chain can be `relayERC20`-ed as long as the source chain is in the dependency set |        |         |
+
+Unit test
+---------
+
+| id    | description                                                                | halmos | echidna |
+| ----- | -----                                                                      | -----  | -----   |
+|   6   | supERC20 token address does not depend on chainID                          |        |         |
```

Comment:
The invariant wouldn't be something like
```suggestion
|   6   | supERC20 token addresses are the same in all the different chains |        |         |
```
?

Reply:
I believe this is currently clearer to map to a halmos proof, like so:

```
check_addressDoesNotDependOnChainId(remote, name, symbol, decimals, chainId1, chainID2){
  vm.assume(chainId1 != chainId2);
  assertEq(factory.deploy(remote, name, symbol, chainId1),  factory.deploy(remote, name, symbol, chainId2));
}
```

wdyt? it certainly _feels_ math-ier

Reply:
ah perfect, if that's simpler for halmos let's keep like that